code/modules/robotics/bot/secbot.dm

```diff
@@ -1185,6 +1198,52 @@ Report Arrests: <A href='?src=\ref[src];operation=report'>[report_arrests ? "On"
 		user.u_equip(W)
 		qdel(W)
 
+	else if (istype(W, /obj/item/baton/) && src.build_step >= 3)
+		if (istype(W, /obj/item/baton/beepsky))	// If we used Beepsky's dropped baton
+			if (src.is_dead_beepsky)							// on Beepsky's corpse
+				boutput(user, "You return Officer Beepsky his trusty baton, reassembling the Securitron! Beep boop.")
+				new /obj/machinery/bot/secbot/beepsky(get_turf(src))
+				qdel(src)
+				user.u_equip(W)
+				qdel(W)
+			else												// On any other securitron assembly?
+				boutput(user, "You give the [src] [W] and connect a cable in the arm to the baton's parallel port, completing the Securitron! Beep boop.")
+				var/obj/machinery/bot/secbot/S = new /obj/machinery/bot/secbot(get_turf(src))
+				S.beacon_freq = src.beacon_freq
+				S.hat = src.hat
+				S.name = src.created_name		// We get an upgraded securitron
+				S.attack_per_step = 2				// 25% chance to attack_on_move, as opposed to 75%
+				S.loot_baton_type = W.type	// So we can drop it all over again.
+				if (W.contraband != 4)
```

Comment:
Using a contraband to check if it's the real deal (I think that's what this is?) is not a great solution especially considering that:
a.) someone might change the baton's contraband level and not know about this bit of code thus breaking it
b.) it's not clear at all what is going on here
I'd probably add a var on the beepsky baton that says if it's the original or a copy.

Reply:
Done!